### PR TITLE
remove enforcementMode from serviceRole proto definition

### PR DESCRIPTION
**Pull request by quanjielin**

### CODE REVIEW

File: rbac/v1alpha1/rbac.proto [hunk 1]
```diff
@@ -108,35 +108,12 @@ package istio.rbac.v1alpha1;
...
   // Required. The set of access rules (permissions) that the role has.
   repeated AccessRule rules = 1;
-
-  // $hide_from_docs
-  // Indicates enforcement mode of the ServiceRole config.
-  // If ServiceRole is in PERMISSIVE mode, all ServiceRoleBindings that
-  // refers to it will be treated as PERMISSIVE mode.
-  // If ServiceRole is in ENFORCED mode, ServiceRoleBindings that
-  // refers to it will decide their own enforcement modes.
-  EnforcementMode mode = 2;
```
[the remark this answers is not in the document]
Comment (diemtvu) [reply rbac/v1alpha1/rbac.proto]: You cannot use the new service role without a (new) binding rule right? (as the rule can bind to one service role only). If that the case, I don't think having this mode in service role helps anything.

[the remark this answers is not in the document]
Comment (yangminzhu) [reply rbac/v1alpha1/rbac.proto]: Well I just realized my concern is not related to this change, but actually related to the local RBAC change as a whole. Simply speaking, in the mixer based RBAC, you could add a new service role without any bindings to disable any access to a service.
But in the local RBAC, you won't be able to do so, at least within the current implementation. (we could try to make the local RBAC to have the same behavior as the mixer based RBAC, but there're more to discuss)

I'll create a different issue to have more discussions on it.

[the remark this answers is not in the document]
Comment (liminw) [reply rbac/v1alpha1/rbac.proto]: As discussed offline. Adding a ServiceRole itself won't take any effect. Only the bindings are the real policies. This should work the same for Mixer based and Envoy based RBAC.

**Pull request merged**